modules/nf-core/modules/ensemblvep/main.nf

```diff
@@ -16,32 +16,36 @@ process ENSEMBLVEP {
     path  extra_files
 
     output:
-    tuple val(meta), path("*.ann.vcf"), emit: vcf
-    path "*.summary.html"             , emit: report
-    path "versions.yml"               , emit: versions
+    tuple val(meta), path("*.ann.vcf")  , optional:true, emit: vcf
+    tuple val(meta), path("*.ann.tab")  , optional:true, emit: tab
+    tuple val(meta), path("*.ann.json") , optional:true, emit: json
+    path "*.summary.html"               , emit: report
+    path "versions.yml"                 , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
+    def vep_output = task.ext.vep_output ?: 'vcf'
+    def suffix = task.ext.suffix ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def dir_cache = cache ? "\${PWD}/${cache}" : "/.vep"
     """
     mkdir $prefix
 
     vep \\
         -i $vcf \\
-        -o ${prefix}.ann.vcf \\
+        -o ${prefix}${suffix}.ann.$vep_output \\
         $args \\
         --assembly $genome \\
         --species $species \\
         --cache \\
         --cache_version $cache_version \\
         --dir_cache $dir_cache \\
         --fork $task.cpus \\
-        --vcf \\
-        --stats_file ${prefix}.summary.html
+        --stats_file ${prefix}.summary.html \\
+        --$vep_output
```

Comment:
For this parameter I would almost say it should be set with `$args` . That way the module output can be configured via the `modules.config` and can also eaily be overwritten by the user, even if we don't parameterize it. (In sarek I would propose an additional parameter though: `--vep_out_format` defaulting to `vcf`

Reply:
I agree with Rike on that